kuma/wiki/tests/test_views.py

```diff
@@ -1208,6 +1208,48 @@ def test_create_on_404(self):
                                            locale=locale))
         eq_(302, response.status_code)
 
+    def test_creating_child_of_redirect(self):
+        """While try to create a child of a redirect,
+        the parent of the child should be redirect's parent"""
+        self.client.login(username='admin', password='testpass')
+        rev = revision(is_approved=True, save=True)
+        doc = rev.document
+        doc_first_slug = doc.slug
+        # Move the document to new slug
+        doc._move_tree(new_slug="moved_doc")
+
+        # Try to create a child with the old slug
+        child_full_slug = doc_first_slug + "/" + "children_document"
+        url = reverse('wiki.document', args=[child_full_slug])
+        response = self.client.get(url, follow=True)
+
+        # The parent id of the query should be same because while moving, a new document is created with old slug
+        # and make redirect to the old document
+        parameters = parse_qs(response.request['QUERY_STRING'])
+        assert parameters['parent'][0] == str(doc.id)
+
+    def test_creating_child_of_redirect_zoned_document(self):
+        """While try to create a child of a redirected zone document,
+           the parent of the child should be redirect's parent"""
+        self.client.login(username='admin', password='testpass')
+        rev = revision(is_approved=True, save=True)
+        root_doc = rev.document
+        # Create a zone of the document
+        zoned_doc = DocumentZone(document=root_doc, url_root="zoned_url")
+        zoned_doc.save()
+
+        # Move the document to new slug
+        root_doc._move_tree(new_slug="moved_doc")
+
+        zoned_child_full_slug = zoned_doc.url_root + "/" + "children_document"
+        response = self.client.get(zoned_child_full_slug, follow=True)
+        print response.status_code
+
+        # The parent id of the query should be same because while moving, a new document is created with old slug
+        # and make redirect to the old document
+        parameters = parse_qs(response.request['QUERY_STRING'])
+        assert parameters['parent'][0] == str(root_doc.id)
```

Comment:
This works, and tests that ``/en-US/zoned_url/children_doc`` finds the right parent. A second test is needed though for the document that now follows a redirect chain into a zone, something like:

```
root_doc = rev.document
original_slug = root_doc.slug
# zone it, move it around, test new page, etc., then:
# test /en-US/docs/2017-04-21-12203442670/child2
unzoned_child_full_slug = original_slug + "/child2"  
response = self.client.get(unzoned_child_full_slug, follow=True)
# ...check status code, parse parameters, and...
assert parameters['parent'][0] == str(root_doc.id)
```
I manually tested, so this should work.